### ISSUE TITLE
Cannot read properties of undefined (reading 'origin')

### ISSUE THREAD

**Issue by rchl** [Adyen/adyen-web#2824]
Our production systems receive a relatively large number (8K a month) of client-side errors due to this code:

https://github.com/Adyen/adyen-web/blob/2f59a65b0e272676164a317c668c5293b870c8a9/packages/lib/src/utils/get-process-message-handler.ts#L19

The code crashes because `event.originalEvent` seems to be `undefined`.

99% of crashes are triggered from Chrome Mobile devices on Android 10.

Suggested fix (adding optional chaining operator):

```
const origin = event.origin || event.originalEvent?.origin;
```

**Comment by ribeiroguilherme** [comment]
Hi @rchl ,

Which version of the adyen-web are you using? Is it happening **only** on Android 10 or also in another versions? And do you know the which browser by any chance?

**Comment by rchl** [comment]
Happen to be using `5.43.0` now but upgraded to `5.68.0` earlier today. Will monitor if it makes any difference but looking at the relevant code, it seems like it will still trigger.

Android 10 collects majority of the crashes but it's also occasionally triggered on other versions (not that not all of that can be trusted since UA can be spoofed):

![Screenshot 2024-09-02 at 14 44 58](https://github.com/user-attachments/assets/9d50b055-9ebe-407b-99d3-cec66c39469a)

As far as user agent it's typically something like

```
Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/127.0.0.0 Mobile Safari/537.36
```

**Comment by ribeiroguilherme** [comment]
@rchl I can see that there are two situations where this code is executed: 
- when the library is initially rendered
- when the 3DS flow is triggered
 
Do you know by any chance in which part of the flow this error happens? And do you have visibility if it is a critical error that crashes the UI/checkout or if it is an uncaught error only?

**Comment by rchl** [comment]
Sorry, I don't have visibility into those details and not sure if it's a critical error that prevents people from completion the transaction.

**Comment by ribeiroguilherme** [comment]
Hi @rchl -  Did you notice any difference in the reports since you upgraded your version?

I ran some tests on Samsung Galaxy S20 with Android 10 using Chrome and unfortunately I didn't manage to reproduce the problem.

**Comment by sponglord** [comment]
@rchl 
It looks to me like something is wrong with the actual `event` being sent (the `event.originalOrigin` stuff is just fallback code for jQuery, and I'd be surprised if it's coming into play here - your'e not using jQuery are you?). 
It's more like `event.origin` just isn't being defined, for some reason.

**Comment by rchl** [comment]
No, I'm not able to determine if this affects payment but I would guess that it doesn't or we'd know about it and we don't include jquery in our pages.

I think that it's just the right thing to null-check the `event.originalOrigin` since it's not part of the [MessageEvent](https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent) event so it's not guaranteed to exist.

**Comment by sponglord** [comment]
Fixed in #2894
